api/v1/services/google_oauth.py

```diff
@@ -0,0 +1,191 @@
+from fastapi import Depends, HTTPException, status
+from datetime import datetime, timezone
+from api.db.database import get_db
+from api.v1.models.user import User
+from api.v1.models.oauth import OAuth
+from api.v1.models.profile import Profile
+from api.core.base.services import Service
+from sqlalchemy.orm import Session
+from typing import Annotated, Union
+from api.v1.services.user import user_service
+from api.v1.schemas.google_oauth import Tokens, UserData, StatusResponse
+
+
+class GoogleOauthServices(Service):
+    """
+    Handles database operations for google oauth
+    """
+    def create(self, google_response: dict,
```

Comment:
toooo many things in this function - we'll refactor it later.

Reply:
Yes Boss.

Thank you